core/src/apps/binance/helpers.py

```diff
@@ -20,11 +20,11 @@
 
 
 def produce_json_for_signing(envelope: BinanceSignTx, msg) -> str:
-    if isinstance(msg, BinanceTransferMsg):
+    if BinanceTransferMsg.is_type_of(msg):
```

Comment:
i expect this to mess mypy up. any way we could get actual `isinstance` to work?

Reply:
Not without quite a bit of work. It would also mean msg objects keep their def alive. Can't we use the typeguard thing?

Reply:
i'll look into it

Reply:
resolved by using typeguards. for that, we need mypy from git though